explorer/ui/app/components/queries/messages.tsx

```diff
@@ -76,33 +59,53 @@ export const xmsgcount = graphql(`
   }
 `)
 
-export const xmsgs: typeof XMsgsDocument = gql(`
-  query XMsgs($cursor: BigInt, $limit: BigInt) {
-    xmsgs(cursor: $cursor, limit: $limit) {
-      TotalCount
-      Edges {
-        Cursor
-        Node {
-          ID
-          StreamOffset
-          SourceMessageSender
-          DestAddress
-          DestGasLimit
-          SourceChainID
-          DestChainID
-          TxHash
-          BlockHeight
-          BlockHash
-          Status
-          SourceBlockTime
-          ReceiptTxHash
-        }
-      }
-      PageInfo {
-        NextCursor
-        PrevCursor
-        HasNextPage
+export const xmsgs = gql(`
+query {
+  xmsgs(first: 10, after: "eyJJRCI6MTksIlBhZ2VOdW0iOjJ9") {
```

Comment:
The after cursor here should not be hard-coded. In the first page it should be missing. Then you get the last item on the first page as the after cursor for the second page. Also initially, you start with backwards pagination to display last items first (i.e. `last: 10`, then `last: 10, before: <first_cursor_of_previous_page>`